sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -456,7 +456,7 @@ class JacksonParser(
             schema.existenceDefaultsBitmask(index) = false
           } catch {
             case e: SparkUpgradeException => throw e
-            case NonFatal(e) if isRoot =>
+            case NonFatal(e) =>
```

Comment:
If you allow this, please, check all complex types `map`, `struct`. Or have you added some tests already?

Reply:
I have tried covering different cases in:
- SPARK-40646: return partial results for JSON arrays with objects
- SPARK-40646: return partial results for objects with values as JSON arrays

I think I am missing a test for map, will add, thanks.

Reply:
I added a test!